src/Bicep.Core/TypeSystem/Providers/ResourceTypesProviderDescriptor.cs

```diff
@@ -11,11 +11,13 @@ public record ResourceTypesProviderDescriptor
     public ResourceTypesProviderDescriptor(
         string name,
         string version,
+        string? artifactRef = null,
         string? alias = null,
         Uri? typesBaseUri = null)
     {
         Name = name;
         Alias = alias ?? name;
+        ArtifactReference = artifactRef ?? "unknown";
```

Comment:
This value is used for error propagation only so its easy to identify from the diagnostic what was the artifact reference that couldn't be resolved, for builtin providers there is no artifact reference  so there is a placeholder value set. Since the scope of this constructor doesn't know how the value is constructed then I chose to use "unknown" as a label, however, this is a provisional change until we normalize the handling of builtin providers and at that time this would become a non-issue. Is this a blocking item?

Reply:
Thanks for having me re-think this.

It's used to propagate the declaration data when an error occurred. The appropriate value should be "{Name}@{Version}" since   This is a reference to a built in provider. However if the import is implicit then maybe "{Name}" is more appropriate...

Reply:
If you choose to keep it, please represent it as a nullable value, rather than a non-null string with a special value ("unknown"). That way you (a) preserve the ability to distinguish between "set" and "not set" without having to rely on string comparisons, and (b) avoid misleading others into assuming it's always going to have a "valid" value.

Reply:
I don't see that you actually changed anything here...?

Reply:
I did , notice that I am setting `ArtifactReference` to either be `name` when the provider is implicit, or `name@version` when the provider is builtin or the reference in all other cases

ArtifactReference = artifactRef ?? (isImplicitImport ? name : $"{name}@{version}");